### PR TITLE
Uior 244 allow user to specify a contributor name type for each contributor added to pol

### PR DESCRIPTION
**Pull request by Feboll**
## Purpose

https://issues.folio.org/browse/UIOR-244

## Approach

Added field contributor on POL form and view.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/POLine/Item/ItemForm.js [hunk 1]
```diff
@@ -174,6 +178,17 @@ class ItemForm extends Component {
               name="details.receivingNote"
             />
           </Col>
+          <Col xs={6}>
+            <Field
```
Comment (NikitaSedyx) [inline src/components/POLine/Item/ItemForm.js]: don't forget that we have templates now and they shouldn't be broken with it

File: src/components/POLine/POLineForm.js [hunk 2]
```diff
@@ -211,6 +212,7 @@ class POLineForm extends Component {
     const funds = getFundsForSelect(parentResources);
     const materialTypes = getMaterialTypesForSelect(parentResources);
     const identifierTypes = getIdentifierTypesForSelect(parentResources);
+    const contributors = getContributorsForSelect(parentResources);
```
Comment (AndreiShumski2) [inline src/components/POLine/POLineForm.js]: IMO it's better to rename `contributors` to `contributorsTypes` or `contributorsNameTypes`

File: src/components/POLine/Item/ItemForm.js (hunk 1 above)
Comment (aliaksei-chumakou) [reply src/components/POLine/Item/ItemForm.js]: @NikitaSedyx do you mean to check if templates work with this change?

Comment (NikitaSedyx) [reply src/components/POLine/Item/ItemForm.js]: yep

File: src/components/POLine/Item/ItemForm.js [hunk 3]
```diff
@@ -174,6 +193,17 @@ class ItemForm extends Component {
...
+              dataOptions={contributorsNameTypes}
+              component={FieldSelection}
+              fullWidth
+              required={contributors.length > 0}
+              label={<FormattedMessage id="ui-orders.itemDetails.contributor" />}
+              name="contributors[0].contributorNameTypeId"
```
Comment (aliaksei-chumakou) [inline src/components/POLine/Item/ItemForm.js]: I don't think it should be `contributors[0]`, we don't use such thing anywhere

File: src/components/POLine/Item/ItemForm.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ import {
   Col,
   Datepicker,
   Row,
+  Select,
```
Comment (aliaksei-chumakou) [inline src/components/POLine/Item/ItemForm.js]: do you use it?

File: src/components/POLine/Item/ItemForm.js [hunk 5]
```diff
@@ -37,12 +38,14 @@ import { isWorkflowStatusOpen } from '../../PurchaseOrder/util';
 
 import css from './ItemForm.css';
 import { ALLOWED_YEAR_LENGTH } from '../const';
+import { FieldSelection } from "@folio/stripes-acq-components";
```
Comment (aliaksei-chumakou) [inline src/components/POLine/Item/ItemForm.js]: please put it above in the right imports order, after line#20

File: src/components/POLine/Item/ItemForm.js [hunk 6]
```diff
@@ -123,11 +126,24 @@ class ItemForm extends Component {
 
     dispatch(change(fieldName, value));
 
+    this.updateContributor();
```
Comment (aliaksei-chumakou) [inline src/components/POLine/Item/ItemForm.js]: this shouldn't be here, all required info is coming from `getInventoryData` above

**Comment by AndreiShumski2** [comment]
+1

**Comment by NikitaSedyx** [comment]
`// TODO: fix the test` :+1: don't forget

**Pull request merged**